src/app/app-canvas/app-canvas.component.ts

```diff
@@ -24,37 +24,58 @@ export class AppCanvasComponent implements OnInit {
 
   AddShapeToCanvas(ObjectToBeRendered: fabric.Object) {
     this.canvas.add(ObjectToBeRendered);
+    let ObjectName = this.CanvasServiceHandler.ObjectName;
     this.EventServiceHandler.addObjectEventMessage(
```

Comment:
The eventMessage should be passed from canvas on object added event, not directly from the addShapeToCanvas function